### PR TITLE
Provide better documentation to p::d::Triangulation::register_data_attach().

### PR DESCRIPTION
**Pull request by bangerth**
While there, also cleanup the code a tiny bit.

This is a first step to document the issue in #6223 a bit better. I think I'll
have to do a similar thing in notify_ready_to_unpack() as well, but that will
have to wait for a separate PR. One step at a time.

### CODE REVIEW

**Comment by bangerth** [comment]
@marcfehling -- FYI.

**Comment by bangerth** [comment]
Ping?

File: include/deal.II/distributed/tria.h [hunk 1]
```diff
@@ -607,39 +607,105 @@ namespace parallel
...
-       * using save() and load() in the same way. Then the status will always
-       * be CELL_PERSIST.
+       * - `CELL_PERSIST`: The cell won't be refined/coarsened, but might be
+       * moved to a different processor. If this is the case, the callback
+       * will want to pack up the data on this cell into an array and store
+       * it at the provided address for later unpacking wherever this cell
+       * may land.
+       * - `CELL_REFINE`: this cell will be refined into 4 or 8 cells (in 2d
+       * and 3d, respectively). However, because these children don't exist
+       * yet, you cannot access them yet at the time when the callback is
```
Comment (kronbichler) [inline include/deal.II/distributed/tria.h]: remove the second 'yet' (?)

**Comment by bangerth** [comment]
@kronbichler -- I fixed the grammar issue. Thanks for the review!

**Pull request merged**